realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -1281,20 +1259,6 @@ private static boolean isClassForRealmModel(Class<?> clazz) {
         }
         throw new IllegalArgumentException("Unexpected value class: " + clazz.getName());
     }
-
-    // TODO: Object Store is not able to merge change set for links list. Luckily since we were still using LinkView
-    // when ship the fine grain notifications, the listener on RealmList is actually added to a OS Results which is
-    // created from the link view. OS Results is computing the change set by comparing the old/new collection. So it
-    // will give the right results if you remove all elements from a RealmList then add all them back and add one more
-    // new element. By right results it means the change set only include one insertion. But if the listener is on the
-    // OS List, the change set will include all ranges of th list. So we keep the old behaviour for
-    // RealmList<RealmModel> for now. See https://github.com/realm/realm-object-store/issues/541
```

Comment:
I tend to agree that this sounds like a mistake, but if we introduce the behaviour (until fixed in OS), then it doesn't seem like a major issue either.